Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridDataConnection.cs

```diff
@@ -981,5 +1020,28 @@ private void UpdateDataProperties()
                 _dataProperties = null;
             }
         }
+
+        private void UpdateIncrementalItemsSource()
+        {
+            // Determine if incremental loading should be used
+            if (_dataSource is ISupportIncrementalLoading incrementalDataSource)
+            {
+                _incrementalItemsSource = incrementalDataSource;
+            }
+            else if (_owner.ItemsSource is ISupportIncrementalLoading incrementalItemsSource)
+            {
+                _incrementalItemsSource = incrementalItemsSource;
+            }
+            else
+            {
+                _incrementalItemsSource = default(ISupportIncrementalLoading);
+            }
+
```

Comment:
Add
```
if (_incrementalItemsSource != null)
{
    INotifyPropertyChanged inpc = _incrementalItemsSource as INotifyPropertyChanged;

    if (inpc != null)
    {
        _weakIncrementalItemsSourcePropertyChangedListener = new WeakEventListener<DataGridDataConnection, object, PropertyChangedEventArgs>(this);
        _weakIncrementalItemsSourcePropertyChangedListener.OnEventAction = (instance, source, eventArgs) => instance.NotifyingIncrementalItemsSource(source, eventArgs);
        _weakIncrementalItemsSourcePropertyChangedListener.OnDetachAction = (weakEventListener) => inpc.PropertyChanged -= weakEventListener.OnEvent;
        inpc.PropertyChanged += _weakIncrementalItemsSourcePropertyChangedListener.OnEvent;
    }
}

```

Reply:
And then you can also add among the private methods above:
```
private void NotifyingIncrementalItemsSource(object sender, PropertyChangedEventArgs e)
{
    if (e.PropertyName == "HasMoreItems")
    {
        _owner.LoadMoreDataFromIncrementalItemsSource();
    }
}

```

What that means is that if the ISupportIncrementalLoading implementation also implements INotifyPropertyChanged and raises PropertyChanged for HasMoreItems, an incremental load may also be triggered.  Thank you!